Sources/TuistKit/Services/TestService.swift

```diff
@@ -60,14 +61,16 @@ final class TestService {
         testServiceGeneratorFactory: TestServiceGeneratorFactorying,
         xcodebuildController: XcodeBuildControlling = XcodeBuildController(),
         buildGraphInspector: BuildGraphInspecting = BuildGraphInspector(),
-        simulatorController: SimulatorControlling = SimulatorController()
+        simulatorController: SimulatorControlling = SimulatorController(),
+        cacheDirectoryProviderFactory: @escaping (Config) -> CacheDirectoriesProviding = { CacheDirectoriesProvider(config: $0) }
```

Comment:
to be more consistent with the codebase, I'd create a new component `final class CacheDirectoryProviderFactory` instead of a closure.